fcrepo-auth-webac/src/main/java/org/fcrepo/auth/webac/WebACFilter.java

```diff
@@ -708,4 +718,42 @@ private static boolean isBinaryOrDescription(final FedoraResource resource) {
                 .anyMatch(t -> t.equals(NON_RDF_SOURCE.toString()) || t.equals(FEDORA_NON_RDF_SOURCE_DESCRIPTION_URI));
     }
 
+    private static boolean isEmbeddedRequest(final HttpServletRequest request) {
+        final var preferTags = request.getHeaders("Prefer");
+        final Set<SinglePrefer> preferTagSet = new HashSet<>();
+        if (preferTags.hasMoreElements()) {
+            preferTagSet.add(new SinglePrefer(preferTags.nextElement()));
+        }
+        final MultiPrefer multiPrefer = new MultiPrefer(preferTagSet);
+        if (multiPrefer.hasReturn()) {
+            final LdpPreferTag ldpPreferences = new LdpPreferTag(multiPrefer.getReturn());
+            return ldpPreferences.prefersEmbed();
+        }
+        return false;
+    }
+
+    private boolean isAuthorizedForContainedResources(final HttpServletRequest request, final Subject currentUser,
+                                                      final FedoraResource resource) {
+        if (!isBinaryOrDescription(resource) && isEmbeddedRequest(request)) {
+            final Transaction transaction = transaction(request);
+            final Stream<FedoraResource> children = resourceFactory.getChildren(TransactionUtils.openTxId(transaction),
+                    resource.getFedoraId());
+            final boolean failures = children.anyMatch(c -> {
+                final URI childURI = URI.create(c.getFedoraId().getFullId());
+                log.debug("Found embedded resource: {}", c);
+                // add the contained URI to the list URIs to retrieve ACLs for
+                addURIToAuthorize(request, childURI);
+                if (!currentUser.isPermitted(new WebACPermission(WEBAC_MODE_READ, childURI))) {
+                    log.debug("Failed to access embedded resource: {}", childURI);
+                    return true;
```

Comment:
Would it make sense to change the logic here such that the collecting variable is named `authorized`, and the final return be `authorized`?
It is slightly counter-intuitive to have "return true" indicating a "failure".

Reply:
It is counter-intuitive, but with the `anyMatch` the first `true` short-circuits the processing immediately. I tried to make a separate Predicate to clean it up, but because I also need access to the `request` for the `addURIToAuthorize` so this was the best way I could find to do it.

Reply:
I take that back, I might be able to make it a little easier to understand.